gtest/DobbyUtilsTest/tests/ContainerIdTest.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <gtest/gtest.h>
+#include "ContainerId.h"
+using namespace std;
+
+class ContainerIdTest : public ::testing::Test {
+         protected:
+                 void SetUp() override {
+                        cout<<"constructor"<<endl;
+                 }
+                 void TearDown() override {
+                        cout<<"destructor"<<endl;
+                 }
+	ContainerId tid,rid;
+};
+
+TEST_F(ContainerIdTest,CheckNumeric){
+        rid = tid.create("123");
+        EXPECT_EQ(rid.str(),"");
+}
+
+TEST_F(ContainerIdTest,CheckDoubleDot){
+        rid = tid.create("a..123");
+        EXPECT_EQ(rid.str(),"");
+}
+
+TEST_F(ContainerIdTest,CheckAlphanumeric){
+        rid = tid.create("a.123");
+        EXPECT_EQ(rid.str(),"a.123");
+}
+
+int main(int argc, char* argv[])
```

Comment:
Why does this test have the `main()` method? 

If we need a main method, we should probably have another file that just act as a test host, instead of coupling the `main()` method to this particular test

Reply:
Removed `main()` method and used the default `gtest_main` library